### ISSUE TITLE
Logging should only use references and not pass values

### ISSUE THREAD

**Issue by orangepaydev** [Jacky26/quarkus-interview-practice#3]
https://github.com/Jacky26/quarkus-interview-practice/blob/add1828e78ebd7695d2dbad8db75b245da747673/src/main/java/org/jc/controller/DataController.java#L63

Any idea what's the impact of doing this?
```
ObjectMappingHelper.toJsonString(jsonResp)
```

Hint
- think about memory consumption?
- think about why we use a templating engine for logging? e.g. why do we this "Successfully get data. Response={}" instead of "Successfully get data. Response=" + ObjectMappingHelper.toJsonString(jsonResp)

**Comment by Jacky26** [comment]
I would say ObjectMappingHelper.toJsonString(jsonResp) will have performance & memory implication as it is calling writeValueAsString() to execute serialization operation on the jsonResp object. It can be an expensive operation in terms of memory as if the size of jsonResp is too large. 

Understand that we used logging framework with templating is to improve the performance by using placeholders instead of concatenation that cause high memory usage. We have to avoid unnecessary string concatenation and creation of string objects.

This is something new for me learn as I haven't facing huge memory impact, good to understand that serialization process will consume additional memory even when we do logging for our application.

**Comment by orangepaydev** [comment]
yes...this could be improved.  Hint only call the serialization only if the log level is at the correct level.

**Comment by Jacky26** [comment]
Yes, we can improve it. I am assuming that "at the correct level." will be the logger.info when we did a success call. For current scenario in my code, it will be under my DataController. I have updated the code, we will remain this line whenever we did a success call, which is a correct log level `logger.info("Successfully get data. Response={}", ObjectMappingHelper.toJsonString(jsonResp));`

Then, I updated the logging in the catch block, will not call the serialization at the ERROR log level, and I update to log the errorCode & error message instead for debugging purpose. 
`logger.error("Failed to get data. ErrorCode=[{}], message=[{}]", e.getErrorCode().getCode(), e.getMessage());` 

Do refer to the latest commit
